sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py

```diff
@@ -2,83 +2,135 @@
 from . import AsyncChallengeAuthPolicy
 from .._user_agent import USER_AGENT
 
 
 if TYPE_CHECKING:
     try:
         # pylint:disable=unused-import
-        from azure.core.credentials import TokenCredential
+        from azure.core.credentials_async import AsyncTokenCredential
+        from ._generated.v7_0 import KeyVaultClient as KeyVaultClientV7
+        from ._generated.v2016_10_01 import KeyVaultClient as KeyVaultClientV2016_10_01
+
+        KeyVaultClientTypes = Union[KeyVaultClientV7, KeyVaultClientV2016_10_01]
     except ImportError:
-        # TokenCredential is a typing_extensions.Protocol; we don't depend on that package
+        # AsyncTokenCredential is a typing_extensions.Protocol; we don't depend on that package
         pass
 
-
-class AsyncKeyVaultClientBase:
+DEFAULT_API_VERSION = V7_0_VERSION
+
+def _get_configuration_class(api_version: str) -> Configuration:
+    """
+    Get the versioned configuration implementation corresponding to the current profile.
+    :return: The versioned configuration implementation.
+    """
+    if api_version == V7_0_VERSION:
+        from ._generated.v7_0.aio._configuration_async import KeyVaultClientConfiguration as ImplConfig
+    elif api_version == V2016_10_01_VERSION:
+        from ._generated.v2016_10_01.aio._configuration_async import KeyVaultClientConfiguration as ImplConfig
+    else:
+        raise NotImplementedError(
+            "API version {} is not supported by this package. ".format(api_version)
+            + "Supported versions are {} and {}".format(V7_0_VERSION, V2016_10_01_VERSION)
+        )
+    return ImplConfig
+
+def _create_config(credential: "AsyncTokenCredential", api_version: str = None, **kwargs: "Any") -> Configuration:
+    if api_version is None:
+        api_version = DEFAULT_API_VERSION
+    config = _get_configuration_class(api_version)(credential, **kwargs)
+    config.authentication_policy = AsyncChallengeAuthPolicy(credential)
+
+    # replace the autorest-generated UserAgentPolicy and its hard-coded user agent
+    # https://github.com/Azure/azure-sdk-for-python/issues/6637
+    config.user_agent_policy = UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
+
+    # Override config policies if found in kwargs
+    # TODO: should be unnecessary after next regeneration (written 2019-08-02)
+    if "user_agent_policy" in kwargs:
+        config.user_agent_policy = kwargs["user_agent_policy"]
+    if "headers_policy" in kwargs:
+        config.headers_policy = kwargs["headers_policy"]
+    if "proxy_policy" in kwargs:
+        config.proxy_policy = kwargs["proxy_policy"]
+    if "logging_policy" in kwargs:
+        config.logging_policy = kwargs["logging_policy"]
+    if "retry_policy" in kwargs:
+        config.retry_policy = kwargs["retry_policy"]
+    if "custom_hook_policy" in kwargs:
+        config.custom_hook_policy = kwargs["custom_hook_policy"]
+    if "redirect_policy" in kwargs:
+        config.redirect_policy = kwargs["redirect_policy"]
+
```

Comment:
Why we need these? They are already implemented in KeyVaultClientConfiguration init.

Reply:
They're vestigial but there's no urgency to remove them. I intend to do it the next time we regenerate the client code.